x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDouble.java

```diff
@@ -42,7 +43,8 @@ public class ToDouble extends AbstractConvertFunction {
         Map.entry(TEXT, ToDoubleFromStringEvaluator.Factory::new),
         Map.entry(UNSIGNED_LONG, ToDoubleFromUnsignedLongEvaluator.Factory::new),
         Map.entry(LONG, ToDoubleFromLongEvaluator.Factory::new), // CastLongToDoubleEvaluator would be a candidate, but not MV'd
-        Map.entry(INTEGER, ToDoubleFromIntEvaluator.Factory::new) // CastIntToDoubleEvaluator would be a candidate, but not MV'd
+        Map.entry(INTEGER, ToDoubleFromIntEvaluator.Factory::new), // CastIntToDoubleEvaluator would be a candidate, but not MV'd
+        Map.entry(EsqlDataTypes.COUNTER_DOUBLE, (field, source) -> field)
```

Comment:
Should we allow `counter_int` and `counter_long` in all of these too?

Reply:
I intended to minimize support for these types, but I'm fine with either way. What's your preference?

Reply:
For consistency it makes sense to add them.

Reply:
ok, I added these in https://github.com/elastic/elasticsearch/pull/107877/commits/84733ff2060dec7600d94fde075432d6522d3433.